src/analysis/raster/qgsrastercalcnode.cpp

```diff
@@ -382,3 +382,42 @@ QgsRasterCalcNode *QgsRasterCalcNode::parseRasterCalcString( const QString &str,
   return localParseRasterCalcString( str, parserErrorMsg );
 }
 
+QStringList QgsRasterCalcNode::referencedLayerNames( const QString &str )
+{
+  QStringList referencedRasters;
+
+  QStringList rasterRef = QgsRasterCalcNode::cleanRasterReferences( str );
+  for ( const auto &i : rasterRef )
+  {
+    if ( referencedRasters.contains( i.mid( 0, i.lastIndexOf( "@" ) ) ) ) continue;
+    referencedRasters << i.mid( 0, i.lastIndexOf( "@" ) );
+  }
+
+  return referencedRasters;
+}
+
+QStringList QgsRasterCalcNode::cleanRasterReferences( const QString &str )
```

Comment:
let's modify this function to be non-static and use "this" object to return its raster references rather than taking an argument:
```c++
QStringList QgsRasterCalcNode::clearRasterReferences()
```

Reply:
Ok, I understand it. 
at this line I should avoid the creation of a qgsrastercalcnode object [https://github.com/qgis/QGIS/blob/d5f14cb8887cba0fa8525860f5e7ad92f7031f01/src/app/qgisapp.cpp#L6578] (https://github.com/qgis/QGIS/blob/d5f14cb8887cba0fa8525860f5e7ad92f7031f01/src/app/qgisapp.cpp#L6578)
and change the check where I use the output of the "ex-static" method? This check [https://github.com/qgis/QGIS/blob/d5f14cb8887cba0fa8525860f5e7ad92f7031f01/src/app/qgisapp.cpp#L6584](https://github.com/qgis/QGIS/blob/d5f14cb8887cba0fa8525860f5e7ad92f7031f01/src/app/qgisapp.cpp#L6584)